Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -121,25 +127,27 @@ ICompilationStatusNotificationPublisher notificationPublisher
       TextDocumentItem textDocument,
       DiagnosticErrorReporter errorReporter,
       Dafny.Program program,
-      bool loadCanceled
+      bool loadCanceled,
+      ILogger logger
```

Comment:
Usually, in .NET Core we inject a logger using the type `ILogger<UserType>`, e.g., `ILogger<TextDocumentLoader`. It's not strictly necessary here since the registration of the factory method is responsible. But this might change in the future when we .NET Core supports factory methods with autowiring.

Moreover, the logger instance used here is passed to `SymbolTable`. Therefore, it'd be better to provide a `ILogger<SymbolTable>` instance here to avoid confusion from the log message. However, since the `TextDocumentLoader` is responsible to create the `SymbolTable` instance, I'd argue that it's the cleanest way to inject an `ILoggerFactory` into the `TextDocumentLoader` and create a suitable instance using `loggerFactory.CreateLogger<SymbolTable>()` to pass to `SymbolTable`.

Reply:
`SymbolTable` are created not only in `TextDocumentLoader`, but also in `SymbolTableRelocator`
I did the a change so that now it guarantees it has an ILogger<SymbolTable> instead of a generic logger.